SU-HSDO/suhumsci
# Feat(STN-562): Splash font animations

# READY FOR REVIEW

## Summary
Add animation to the "Splash font" text style.

**Colorful Theme:**
Fade in (opacity) and scale upon load.

**Traditional Theme:**
Fade in (opacity)

Codepen Examples:
https://codepen.io/heatherdesigns/pen/NWjBqob?editors=1100

JIRA Card:
https://sparkbox.atlassian.net/browse/STN-562

## Need Review By (Date)
9/29

## Urgency
medium

## Steps to Test
1. npm run test
2. Create a Flexible Page with a Text Area with Splash Font text style added. 
- Add a Text Area.
- Add some text and highlight the text
- Use the "Style" dropdown and select "Splash Font"
- This will apply the hb-splash-font class to the text when you save.
3. To see the animations you need to make sure the "Animation Enhancements" needs to be turned on on the Colorful and/or Traditional theme settings. The animations will not work if this is not enabled.
4. You should also test the Callout Box paragraph component with Splash Font style as well, using the similar steps above, you would add a "Callout Box" instead of Text Area and test it with both floating left and right on the page.
- _**Note:** Found a bug created by a css injector style sheet that is adding -10% margin to left and right aligned Callout Boxes, which is pushing the callout box off the edge of the content areas. This is a bug we will surface with H&S and is not part of this work._
5. The animation should only play once as you scroll down the page. It should not play again if you scroll back up.

**The animation should not work when these settings are set:**
- The animation setting is turned off for the Humsci theme setting:
![image](https://user-images.githubusercontent.com/9355698/135181279-d5b77b8f-ef2a-44b1-bbfb-20a797355485.png)
- The animation will not play if JavaScript is disabled. You can target the no-js class in the <html> and set the opacity of hs-font-splash to 1 to ensure the text displays when JavaScript is disabled.
- The animation should not play if a user has their system settings for `prefers-reduced-motion` set to reduce. Read more about `prefers-reduced-motion `on MDN and CSS Tricks.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
